src/LinAlg/Test/MatrixTests.h

```diff
@@ -10,54 +10,62 @@
 //!
 //==============================================================================
 
-#include "MatVec.h"
+#include "matrixnd.h"
 
-#include "gtest/gtest.h"
+#include <gtest/gtest.h>
 
 #include <limits>
 #include <numeric>
 
+
 namespace {
 
 template<class Scalar>
 void vectorAddTest()
 {
-  constexpr size_t size = 10;
-
-  utl::vector<Scalar> d(size);
-  std::iota(d.begin(), d.end(), 0.0);
+  utl::vector<Scalar> d(12);
+  std::iota(d.begin(), d.end(), 1.0);
 
   utl::vector<Scalar> d2(d);
   d2.add(d, 1.5);
+  std::cout <<"d2:"<< d2 << std::endl;
 
-  utl::vector<Scalar> d3(size / 2);
+  utl::vector<Scalar> d3(d.size() / 2);
   d3.add(d, 0.5, 1, 2);
+  std::cout <<"d3:"<< d3 << std::endl;
 
-  utl::vector<Scalar> d4(size);
+  utl::vector<Scalar> d4(d.size());
   d4.add(d3, 1.0, 0, 1, 0, 2);
   d4.add(d3, 1.0, 0, 1, 1, 2);
-  std::cout << d4 << std::endl;
+  std::cout <<"d4:"<< d4 << std::endl;
 
-  for (size_t i = 0; i < size; ++i) {
-    EXPECT_FLOAT_EQ(d2[i], 2.5*i);
+  for (size_t i = 0; i < d2.size(); ++i) {
+    EXPECT_FLOAT_EQ(d2[i], 2.5*(i+1));
     EXPECT_FLOAT_EQ(d4[i], d3[i/2]);
   }
-  for (size_t i = 0; i < size / 2; ++i)
-    EXPECT_FLOAT_EQ(d3[i], i + 0.5);
 
-  if constexpr (std::is_same_v<Scalar,Real>) {
-    utl::vector<Scalar> v = d + d;
-    for (size_t i = 0; i < size; ++i)
-      EXPECT_FLOAT_EQ(v[i], 2.0*i);
-  }
+  for (size_t i = 0; i < d3.size(); ++i)
+    EXPECT_FLOAT_EQ(d3[i], i+1);
+
+  d2.fill(0.0);
+  d2.add(d, 1.0, 0, 3, 0, 3);
+  d2.add(d, 1.0, 1, 3, 1, 3);
+  d2.add(d, 1.0, 2, 3, 2, 3);
+  std::cout <<"d5:"<< d2 << std::endl;
+  for (size_t i = 0; i < d.size(); ++i)
+    EXPECT_FLOAT_EQ(d[i], d2[i]);
+  d2.add(d, 1.0, 0, 1, 7, 3);
+  std::cout <<"d6:"<< d2 << std::endl;
+  for (size_t i = 7; i < d.size(); i += 3)
+    EXPECT_FLOAT_EQ(d2[i], d[i]+(i-4)/3);
 }
 
 
 template<class Scalar>
 void vectorDotTest()
 {
   constexpr size_t size = 10;
-  constexpr Scalar max = Scalar(size-1) * Scalar(size) / 2.0;
+  constexpr Scalar max = Scalar((size-1)*size/2);
```

Comment:
if you cast after the division you're doing the division as integer.

Reply:
yes, and that is totally fine here, since `(size-1)*size` always will be an even number no-matter the value on `size` such that dividing by two will/should always result in the same as when using float arithmetics